### PR TITLE
[backend] check playbook filters & add playbook nodes tests (#8721)

### LINKED ISSUE #8721: [Playbooks] unable to filter on "Platform Creation Date" on "Query knowledge on regular basis"
## Description

Error messaged received when trying to create a Playbook that "Query knowledge on regular basis" with a "Platform Creation Date" filter -> receives an Error (see below)

## Environment

OpenCTI version: 6.3.6

## Reproducible Steps

1. Go to Data > Processing > Create a new playbook
2. Click the rectangle to create the first step of the playbook and choose ""Query knowledge on regular basis"
3. Fill in the info and add the filter "Platform creation date" with a date

Example: 

Screenshot 2024-10-21 at 14 12 32

## Expected Output

Should filter results from the date noted in the filter

## Actual Output

Error message: 

Screenshot 2024-10-21 at 14 12 12

### PR DESCRIPTION
**Pull request by Archidoit**
### Proposed changes
Modify the playbook filters checking done in backend at playbook node adding/replacement.

### Related issues
https://github.com/OpenCTI-Platform/opencti/issues/8721

### CODE REVIEW

**Comment by SouadHadjiat** [comment]
would it be possible to add a test for this bug fix ? a playbook-test is now available on master with resolver tests that can be completed.

File: opencti-platform/opencti-graphql/src/modules/playbook/playbook-domain.ts [hunk 1]
```diff
@@ -79,15 +80,25 @@ export const getPlaybookDefinition = async (context: AuthContext, playbook: Basi
   return playbook.playbook_definition;
 };
 
-export const playbookAddNode = async (context: AuthContext, user: AuthUser, id: string, input: PlaybookAddNodeInput) => {
-  // our stix matching is currently limited, we need to validate the input filters
+const checkPlaybookFiltersAndBuildConfigWithCorrectFilters = (input: PlaybookAddNodeInput) => {
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/modules/playbook/playbook-domain.ts:83]: this method can also be unit tested

Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/modules/playbook/playbook-domain.ts:83]: tests added :)

File: opencti-platform/opencti-graphql/tests/utils/testQueryHelper.ts [hunk 2]
```diff
@@ -31,10 +31,31 @@ export const adminQueryWithSuccess = async (request: { query: any, variables: an
     variables: request.variables,
   });
   expect(requestResult, `Something is wrong with this query: ${request.query}`).toBeDefined();
+  console.log('requestResult.errors', requestResult.errors);
```
Comment (lndrtrbn) [inline opencti-platform/opencti-graphql/tests/utils/testQueryHelper.ts]: Does the console.log is intended?

Comment (Archidoit) [reply opencti-platform/opencti-graphql/tests/utils/testQueryHelper.ts]: noooo gonna remove it sorry

**Review by lndrtrbn — COMMENTED** [review]
Works fine, I restarted the CI, e2e were failing

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**